tests/testdata/qgis_server/wfs_getfeature_within3857FilterTemplate_post.txt

```diff
@@ -0,0 +1,60 @@
+Content-Type: text/xml; subtype=gml/2.1.2; charset=utf-8
```

Comment:
Before I dig into this again, would you please confirm that this is the expected output for the case in which you are asking for  3857 features and a 3857 within filter? 

Because it seems to me that you are expecting `8.20349634,44.90148253 8.20349634,44.90148253` 4326 coordinates, should it not be in 3857?

Reply:
Hi @elpaso 
No, this file is not the expected output. 

I only know that the output is wrong since there is no feature returned although the polygon is the same as in 4326. Since I don't know the exact expected output, I just put the same file as in the test 4326 request. 

Or is there a way to get the correct output besides doing the request?

Reply:
I would keep things as simple as possible: you can forge a request around a single feature, just make a tiny buffer around the point and take the BBOX.
Then you will expect just that feature back with its coordinates (I'm sure you know how to convert them from 4326 to 3857). You can probably take the 4326 and change the coords and srs.

Reply:
I have updated the test, but I am still not sure if the converted polygon falls correctly on the expected point, I will check again tomorrow with a more clear head.

Reply:
Thank you @ismailsunni and sorry if I wasn't clear. I see that this PR still contains a set of test cases that compare output calls to verify that they are **not** equal to a reference file.

What I am asking you to provide is a singe failing test case (one: not many) that given a request (if I understand what you are trying to demonstrate, a request with a GML 3857 encoded within filter and 3857 features result) produces an expected reference output. 

In other words a "positive" (but failing) test with an expected output (it does not matter if it's forged by hand).

Is that clear?

If you want you can leave the other (just one) positive test that already passes (the one with 4326) for reference, but that is not really required.

Reply:
Hi @elpaso

> I see that this PR still contains a set of test cases that compare output calls to verify that they are not equal to a reference file.

Which one is it? I only add 2 unit tests in this PR, first for 4326 references, and the other one for 3857 with output result created by hand as you suggested yesterday.

I understand what you mean actually, that's why I removed the third unit test from the reverted PR on this PR.

Now, I think I have good unit test, yesterday I was doing silly thing and messed up the polygon.

As a reference, this is the polygon (as 4 blue points) with the coordinates on 4326 (red) and 3857 (black).
![image](https://user-images.githubusercontent.com/1421861/84123206-cef14e80-aa39-11ea-80ee-ef6edbe0c745.png)

> If you want you can leave the other (just one) positive test that already passes (the one with 4326) for reference, but that is not really required.

If it's ok, I want to keep it, for reference 

Thanks @elpaso! Let me know if you need anything else.